src/openms/include/OpenMS/ANALYSIS/ID/PeptideIndexing.h

```diff
@@ -677,8 +699,155 @@ namespace OpenMS
+             String seq_lower = seq;
+             seq_lower.toLower();
+
+             // remove all non-alphanumeric characters
+             seq_lower.erase(std::remove_if(seq_lower.begin(), seq_lower.end(),
+                                            [](char c) { return !std::isalpha(c); } ), seq_lower.end());
+
+             // search for case insensitive occurrence of the possible decoys
+             for (auto pair : decoy_count)
+             {
+               if (seq_lower.hasPrefix(pair.first))
+               {
+                 // count observed (ignoring case)
+                 decoy_count[pair.first].first++;
+                 // store observed (case sensitive)
+                 std::string seq_decoy = StringUtils::prefix(seq, pair.first.length());
+                 decoy_case_sensitive[pair.first] = seq_decoy;
+               }
+               else if (seq_lower.hasSuffix(pair.first))
+               {
+                 // count observed (ignoring case)
+                 decoy_count[pair.first].second++;
+                 // store observed (case sensitive)
+                 std::string seq_decoy = StringUtils::suffix(seq, pair.first.length());
+                 decoy_case_sensitive[pair.first] = seq_decoy;
+               }
+             }
+           }
+         }
+       }
+
+       // DEBUG only: print counts of found decoys
+       for (auto &a : decoy_count) LOG_DEBUG << a.first << "\t" << a.second.first << "\t" << a.second.second << std::endl;
+
+       return determineDecoyStringFromCounts_(decoy_count, decoy_case_sensitive, decoy_string, is_prefix);
+     }
+
+     ExitCodes determineDecoyStringFromCounts_(const MapDecoyStringToPrefixSuffixCount & decoy_count,
+                                     const MapCaseInsensitiveToCaseSensitiveDecoyString & decoy_case_sensitive,
+                                     std::string & decoy_string,
+                                     bool & is_prefix) const
+     {
+       int all_prefix_occur(0), all_suffix_occur(0);
+
+       // count observed decoy prefixes
+       for (const auto &pair : decoy_count)
+       {
+         all_prefix_occur += pair.second.first;
+       }
+
+       // count observed decoy suffixes
+       for (const auto &pair : decoy_count)
+       {
+         all_suffix_occur += pair.second.second;
+       }
+
+       if (all_prefix_occur == all_suffix_occur)
+       {
+         LOG_ERROR << "Unable to determine decoy string" << std::endl;
+         return DECOYSTRING_EMPTY;
+       }
+
+       // if a decoy prefix occured at least 80% -> set it as prefix decoy
+       for (const auto & pair : decoy_count)
+       {
+         const std::string & case_insensitive_decoy_string = pair.first;
+         const std::pair<int, int> & prefix_suffix_counts = pair.second;
+         double freq_prefix = static_cast<double>(prefix_suffix_counts.first) / static_cast<double>(all_prefix_occur);
+         if (freq_prefix > 0.8)
```

Comment:
Does this mean, that finding a single random decoy string once, will make it the winner?
Mhh... usually one has 50% decoys + 50% foward, so there might be a better heuristic using the total protein counts?

Reply:
I agree, you could change this to > 40% of the total proteins carrying the string (and the string being the most common one across all of them)

Reply:
Since we're iterating over the decoy_count here, the first decoystring with > 80% occurence wins. So, the first random decoy string shouldn't just win, it would also have to satisfy the condition:
>80% of all pre/suf occurences of the decoy string are of this string

Just checking prefixes first here. 
However, if you guys feel like the total protein counts result in a more accurate heuristic - sure!